easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -898,7 +898,12 @@ def dump(self, fp, always_overwrite=True, backup=False):
             if self.template_values[key] not in templ_val and len(self.template_values[key]) > 2:
                 templ_val[self.template_values[key]] = key
 
-        ectxt = self.parser.dump(self, default_values, templ_const, templ_val)
+        try:
+            ectxt = self.parser.dump(self, default_values, templ_const, templ_val)
+        except NotImplementedError as err:
+            self.enable_templating = orig_enable_templating
+            raise NotImplementedError(err)
```

Comment:
@ocaisa: Can you clarify why this is needed? It seems useless since you're re-raising the error anyway?

Reply:
This error is being caught in a try/except for yeb files and turned into a log error, if the templating value is not restored it results in a problem when doing anything with the ec instance since templating is switched off otherwise.

Reply:
To clarify, I `reproduce_build()` catches this error and turns it into a warning. It then continues with the build (which can still succeed) but requires that the original `enable_templating` value was restored.

Reply:
I've added a comment that hopefully makes this a bit clearer